### ISSUE TITLE
Implement or Improve Concept Exercise: object-destructuring

### ISSUE THREAD

**Issue by SleeplessByte** [exercism/javascript#948]
## Getting started
Here you can read about what Concept Exercises are and how they are structured:

- [What are Concept Exercises?](https://exercism.org/docs/building/product/concept-exercises)
- [Concept Exercises](https://exercism.org/docs/building/tracks/concept-exercises)
- [Concepts](https://exercism.org/docs/building/tracks/concepts)

If you have not done so yet, it is probably also helpful to do a couple of "Learning Exercises" (this is how they are called on the site) yourself. You can also look at the code of an existing concept exercise like [`bird-watcher`](https://github.com/exercism/javascript/tree/main/exercises/concept/bird-watcher) (concept [`for-loops`](https://github.com/exercism/javascript/tree/main/concepts/for-loops)) for reference.

See the documentation _above_ (general documentation), as well as [How to implement a Concept Exercise in JavaScript](https://github.com/exercism/javascript/blob/main/reference/implementing-a-concept-exercise.md).

## Goal

The goal of this exercise is to teach the student how they can _destructure_ an object into multiple bindings (which looks like pattern matching), concept `object-destructuring`.

## Learning objectives

- Using destructuring to get a property
- Using destructuring to get a nested property
- Using destructurng to get multiple properties ([object pattern shorthand](https://exploringjs.com/impatient-js/ch_destructuring.html#property-value-shorthands))
- Using destructuring to _rename_ properties
- Using destructuring + [rest properties](https://exploringjs.com/impatient-js/ch_destructuring.html#rest-properties) to extract a property, and put the rest of the properties in a new name (after.md: React has the pattern  `...otherProps` to pass on a few props, but not all).
- Using [default values in object destructuring](https://exploringjs.com/impatient-js/ch_destructuring.html#default-values-in-object-destructuring)
- Using [nested destructuring](https://exploringjs.com/impatient-js/ch_destructuring.html#nested-destructuring).

## Out of scope

- Anything _specific_ with arrays because there is another concept for this

## Concepts

- `object-destructuring`

## Prerequisites

- `objects`
- `array-destructuring`
- ... (do you think something is missing?)

### About

- Usage in the parameters directly,
- Usage when extracting data from e.g. a JSON response
- Usage in e.g. React

## Help

You can choose to do this solo-style, or collaborate with multiple people on this. The suggested approach is to

1. **First accept this issue by saying "I'd like to work on this"** (no need to wait for a response, just go with it) and optionally request that someone works with you (and wait for a second person to accept your request).
2. Use this issue to discuss any questions you have, what should be included in the content and what not and to collect reference material.
3. Create a PR and set "exercism/javascript" as reviewers. Additionally you can write in #maintaining-javascript that your PR is ready for review. Once you incorporated any critical feedback that the reviewer might give you and the PR is approved, it will be merged by a maintainer.

**Comment by junedev** [comment]
Before starting with this, please have a look at the existing `rest-and-spread` and `array-destructering` concept and the `elyses-destructured-enchantments` exercise. An idea is that the learning objectives mentioned in this issue could be integrated into the existing exercise so the student would see all the rest/spread and destructering action in one place. The concept `array-destructering` could be renamed to `destructering` for this purpose. This is up for discussion. Just an idea how to not create too many too small concepts / too many concept exercises.

**Comment by junedev** [comment]
This is on hold. The description needs to be updated according to the structure described here: https://github.com/exercism/javascript/issues/1500  (I will probably just create a new issue.)

**Comment by junedev** [comment]
There is a fresh issue for this now (https://github.com/exercism/javascript/issues/1501), I am closing this one.
